effekt/jvm/src/main/scala/effekt/Runner.scala

```diff
@@ -42,15 +42,44 @@ trait Runner[Executable] {
   /**
    * Creates a OS-specific script file that will execute the command when executed,
    * forwarding command line arguments.
+   * `$SCRIPT_DIR` refers to the directory the script is in.
+   * 
    * @return the actual name of the generated script (might be `!= name`)
    */
   def createScript(name: String, command: String*): String = os match {
     case OS.POSIX =>
-      IO.createFile(name, s"#!/bin/sh\n${command.mkString(" ")} \"$$@\"", true)
+      val computeScriptDir =
+        """# Determine the directory of the script
+          |if [ -n "$BASH_SOURCE" ]; then
+          |    # Bash
+          |    SCRIPT_DIR=$(dirname "$BASH_SOURCE")
+          |elif [ -n "$ZSH_VERSION" ]; then
+          |    # Zsh
+          |    SCRIPT_DIR=$(dirname "$0")
+          |elif [ -n "$KSH_VERSION" ]; then
+          |    # Ksh
+          |    SCRIPT_DIR=$(dirname "${.sh.file}")
+          |else
+          |    # POSIX shells (sh, dash, etc.)
+          |    SCRIPT_DIR=$(dirname "$0")
+          |fi
+          |""".stripMargin
```

Comment:
Do we need this? Can't we just always use `SCRIPT_DIR=$(dirname "$0")`?
(I _really_ don't know enough cross-platform `sh` to be honest)

Reply:
(me neither, so I just kept it 😅)
Probably we don't and you're right 🤔

Reply:
A potential downside might be symlinks: `$BASH_SOURCE` should supposedly have the "true" source as opposed to `$0`

Reply:
I first thought "do we care?" - but we might if we want to package an Effekt-application with nix, since it might just symlink the one file (and arguably should only symlink that) 🤔

Reply:
But then, `BASH_SOURCE` is not portable and in this case we might want to use e.g. `realpath`